pkg/storage/formattedstore/batchwrite.go

```diff
@@ -0,0 +1,160 @@
+}
+
+// BatchService is a batch service.
+type BatchService struct {
+	sizeLimit int
+	timeLimit time.Duration
+	sync.RWMutex
+	values       models.Batch
+	futureValues map[string]*batch
+	formatter    Formatter
+	provider     batchProvider // TODO remove batch provider after refactoring edv batch
+}
+
+type batch struct {
+	keyID     string
+	value     []byte
+	isDeleted bool
+}
+
+// NewBatchWrite new batch write.
+func NewBatchWrite(sizeLimit int, timeLimit time.Duration, formatter Formatter,
+	provider batchProvider) *BatchService {
+	ticker := time.NewTicker(timeLimit)
+	quit := make(chan struct{})
+	b := &BatchService{
+		sizeLimit: sizeLimit, timeLimit: timeLimit, values: make(models.Batch, 0),
+		futureValues: make(map[string]*batch), formatter: formatter, provider: provider,
+	}
+
+	go func(b *BatchService) {
+		for {
+			select {
+			case <-ticker.C:
+				if err := b.flush(); err != nil {
+					logger.Errorf(err.Error())
+				}
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}(b)
+
+	return b
+}
+
+// Get value from batch values.
+func (b *BatchService) Get(k string) ([]byte, error) {
+	b.RLock()
+	v, ok := b.futureValues[k]
+	b.RUnlock()
+
+	if !ok {
+		return nil, fmt.Errorf("k %s not found", k)
+	}
+
+	if v.isDeleted {
+		return nil, ErrValueIsDeleted
+	}
+
+	return v.value, nil
+}
+
+// Put in batch value.
+func (b *BatchService) Put(s batchStore, k string, v []byte) error {
+	b.RLock()
+	size := len(b.values)
+	b.RUnlock()
+
+	if size >= b.sizeLimit {
+		if err := b.flush(); err != nil {
+			return err
+		}
+	}
+
+	b.Lock()
+	b.futureValues[k] = &batch{keyID: k, value: v, isDeleted: false}
+	b.Unlock()
+
+	go func() {
+		formattedValue, err := b.formatter.FormatPair(k, v)
+		if err != nil {
+			logger.Errorf("failed to format pair for k=%s: %w", k, err)
+			return
+		}
+
+		encryptedDocument, err := s.AddEncryptedIndices(k, formattedValue)
+		if err != nil {
+			logger.Errorf("failed to add encrypted indices k=%s: %w", k, err)
+			return
+		}
+
+		b.Lock()
+		b.values = append(b.values, models.VaultOperation{
+			Operation:         models.UpsertDocumentVaultOperation,
+			EncryptedDocument: *encryptedDocument,
+		})
+		b.Unlock()
+	}()
+
+	return nil
+}
+
+// Delete value.
+func (b *BatchService) Delete(k string) error {
+	b.Lock()
+	b.futureValues[k] = &batch{keyID: k, isDeleted: true}
+	b.values = append(b.values, models.VaultOperation{
+		Operation:  models.DeleteDocumentVaultOperation,
+		DocumentID: b.formatter.GenerateEDVCompatibleID(k),
+	})
+	b.Unlock()
+
+	return nil
+}
+
+func (b *BatchService) flush() error {
+	start := time.Now()
+
+	b.Lock()
+	if len(b.values) != 0 {
+		// call edv server to commit the batch
+		if err := b.provider.Batch(&b.values); err != nil {
+			return err
+		}
+
+		b.values = make(models.Batch, 0)
+		b.futureValues = make(map[string]*batch)
+	}
+	b.Unlock()
+
+	logger.Errorf("flush take time %s", time.Since(start))
```

Comment:
This message is actually really useful for me doing EDV server performance testing. Can you change it to `logger.Infof`?

Also, can you print out the time it takes to do just the`b.provider.Batch(&b.values)` method? I can always do this later, but it will save me time if you just add it now since I'll need it constantly (and this way I don't have to run local aries-framework-go version)